src/common/redis_module/chain_module.h

```diff
@@ -0,0 +1,50 @@
+#ifndef RAY_CHAIN_MODULE_H_
+#define RAY_CHAIN_MODULE_H_
+
+#include <functional>
+
+#include "redismodule.h"
+
+// NOTE(zongheng): this duplicated declaration serves as forward-declaration
+// only.  The implementation is supposed to be linked in from credis.  In
+// principle, we can expose a header from credis and simple include that header.
+// This is left as future work.
+
+// Typical usage to make an existing redismodule command chain-compatible:
+//
+//   extern RedisChainModule module;
+//   int MyCmd_RedisModuleCmd(...) {
+//       return module.Mutate(..., NodeFunc, TailFunc);
+//   }
+//
+// See, for instance, ChainTableAdd_RedisCommand in ray_redis_module.cc.
+class RedisChainModule {
```

Comment:
Not necessary for this PR, but a suggestion to make it easier to port over other commands in the future is to define another module that implements the common interface, but is for non-chain-replicated commands, like `RedisModule`. Then, I believe we could do something like this in `ray_redis_module.cc`:

```
#if RAY_USE_NEW_GCS
  module = RedisChainModule;
#else
  module = RedisModule;
#endif

  if (RedisModule_CreateCommand(ctx, "ray.table_add",
                                module.Mutate(TableAdd_DoWrite, TableAdd_DoPublish), "write pubsub", 0,
                                0, 0) == REDISMODULE_ERR) {
    return REDISMODULE_ERR;
  }
```